### ISSUE TITLE
F401 false positive when import is used in stringified generic subscript

### ISSUE THREAD

**Issue by sterliakov** [astral-sh/ruff#10676]
Keywords: `F401`, "unused import", "generics"
This closely resembles #9121 

## Reproducer:

```python
from typing import TYPE_CHECKING, Generic, TypeVar
    
if TYPE_CHECKING:
    from uuid import UUID

_T = TypeVar('_T')

class Base(Generic[_T]): ...
class Derived(Base["UUID"]): ...
```

Output:
```
$ ruff check --isolated a.py
a.py:4:22: F401 [*] `uuid.UUID` imported but unused
Found 1 error.
[*] 1 fixable with the `--fix` option.
```

## Configiration

Ruff settings: reproduces without `pyproject.toml` on default settings.
Ruff version: `0.3.4` installed from PyPI

## Notes

Autofixer removes this import, obviously causing `mypy` error (`name "UUID" is not defined [name-defined]`) later.

**Comment by charliermarsh** [comment]
We don't currently have any logic to infer whether an arbitrary subscript operations is a generics, but we expose an `extend-generics` setting that you can use to tell Ruff to treat `Base` as such.

**Comment by sterliakov** [comment]
Ough, sorry, this ticket is anyway a duplicate of #9298, I'll post my further suggestion there instead and close this.
